slatedb/src/mem_table.rs

```diff
@@ -113,12 +113,17 @@ pub(crate) struct WritableKVTable {
 }
 
 pub(crate) struct ImmutableMemtable {
-    /// The last WAL ID of this IMM. This is used to determine the starting position of
-    /// WAL replay during recovery. After an IMM is flushed to L0, we do not need to care
-    /// about the earlier WALs which produced this IMM, all we need to know is the last
+    /// The recent flushed WAL ID when this IMM is freezed. This is used to determine the starting
+    /// position of WAL replay during recovery. After an IMM is flushed to L0, we do not need to
+    /// care about the earlier WALs which produced this IMM, all we need to know is the recent
     /// WAL ID of the last L0 compacted.
-    last_wal_id: u64,
+    ///
+    /// Please note that this recent flushed WAL ID might not exactly match the last WAL ID that
```

Comment:
bit of a nitpick but I'm not really getting why we're loosening this name. It's meant to represent the highest wal id for which all of the data in the wal object is present in the memtable. `last_wal_id` feels more accurate than `recent_flushed_wal_id` which just implies that it's some recent wal id that this memtable has all the data for.

Reply:
this might related with https://github.com/slatedb/slatedb/pull/519, in #519, it tries to decouple the flushing of WALs as its own schedule (until max_flush_bytes / max_flush_seconds reached or manually triggering `flush()`), as a result, when a memtable freezed to imm, the related WAL entries may still be buffered in the memory without `wal_id` assigned, so all it could get is the `recent_flushed_wal_id`, which is helpful to navigate to the start position for WAL replaying on recovery.